### PR TITLE
New arm control

### PR DESCRIPTION
**Pull request by ipa-nhg**

### CODE REVIEW

**Comment by ipa-nhg** [comment]
news here? tested and working?

@lokalmatador please solve merge conflict

**Comment by ghost** [comment]
Kind of: position control should be mostly done, but as UIBK's arm is not working anymore (it seems one motor is broken), expect some delay...

File: squirrel_control/src/squirrel_hw_interface.cpp [hunk 1]
```diff
@@ -0,0 +1,508 @@
...
+    default:
+      throw_control_error(true, "Unknown mode: " << current_mode_);
+    }
+  }
+
+
+  void SquirrelHWInterface::write(ros::Duration &elapsed_time) {
+    //This for that the robot keeps its initial pose and does not move back to 0 (deault initialization of C++ for class members)
+    if (hold) {
+      joint_position_command_ = joint_position_;
```
Comment (NadiaHG) [inline squirrel_control/src/squirrel_hw_interface.cpp]: these lines are not been "used", shouldn't be out of the if loop?

Comment (ghost) [reply squirrel_control/src/squirrel_hw_interface.cpp]: They are, but just once. When starting the controller and creating the HW interface "hold" is initialized to true. Thus, when executing write() for the first time after being launched the controller uses the current joint values to initialize the joint_position_command_ (as of being a class member C++ initializes all values to 0.0) for that the arm stays in its initial pose - this is really important - imagine the controller launching and crushing the hull as of suddenly moving to its zero position. Anyways, after being executed once, "hold" is set to false and this if is never executed again. Maybe there exists a nicer solution, but for now this works just perfect. This is actually also were we set all base values to 0.0 but the arm just moves.

**Comment by ipa-nhg** [comment]
new control is already developed and tested: I would like to merge this pull request, will be done through the PR https://github.com/squirrel-project/squirrel_driver/pull/112 where the merge conflict is solved
Please for further issues or features open new issues

**Comment by ghost** [comment]
Position control is now working. There are just some issues with the base limits we should fix, i.e., enforceLimits() for now removes all base commands as they apparently are not within the limits specified in the URDF - this could also be caused by the base only accepting velocities and no positions which is why this information probably is not inside the URDF? We can have a chat if you want!

**Comment by ipa-nhg** [comment]
@lokalmatador new issue to continue the discussion https://github.com/squirrel-project/squirrel_driver/issues/113

**Pull request merged**